ext/har/types.go

```diff
@@ -0,0 +1,365 @@
+		harNameValuePair := NameValuePair{
+			Name:  escapedKey,
+			Value: escapedValues,
+		}
+		harQueryString[index] = harNameValuePair
+		index++
+	}
+	return harQueryString
+}
+
+func parseCookies(cookies []*http.Cookie) []Cookie {
+	harCookies := make([]Cookie, len(cookies))
+	for i, cookie := range cookies {
+		harCookie := Cookie{
+			Name:     cookie.Name,
+			Domain:   cookie.Domain,
+			HttpOnly: cookie.HttpOnly,
+			Path:     cookie.Path,
+			Secure:   cookie.Secure,
+			Value:    cookie.Value,
+		}
+		if !cookie.Expires.IsZero() {
+			harCookie.Expires = &cookie.Expires
+		}
+		harCookies[i] = harCookie
+	}
+	return harCookies
+}
+
+type Response struct {
+	Status      int             `json:"status"`
+	StatusText  string          `json:"statusText"`
+	HttpVersion string          `json:"httpVersion"`
+	Cookies     []Cookie        `json:"cookies"`
+	Headers     []NameValuePair `json:"headers"`
+	Content     Content         `json:"content"`
+	RedirectUrl string          `json:"redirectURL"`
+	BodySize    int64           `json:"bodySize"`
+	HeadersSize int64           `json:"headersSize"`
+	Comment     string          `json:"comment,omitempty"`
+}
+
+func ParseResponse(resp *http.Response, captureContent bool) *Response {
+    if resp == nil {
+        return nil
+    }
+
+    statusText := resp.Status
+    if len(resp.Status) > 4 {
+        statusText = resp.Status[4:]
+    }
+    redirectURL := resp.Header.Get("Location")
+    harResponse := Response{
+        Status:      resp.StatusCode,
+        StatusText:  statusText,
+        HttpVersion: resp.Proto,
+        Cookies:     parseCookies(resp.Cookies()),
+        Headers:     parseStringArrMap(resp.Header),
+        RedirectUrl: redirectURL,
+        BodySize:    resp.ContentLength,
+        HeadersSize: calcHeaderSize(resp.Header),
+    }
+
+    if captureContent && resp.Body != nil {
+        body, err := io.ReadAll(resp.Body)
+        if err != nil {
+            log.Printf("Error reading response body: %v", err)
+            return &harResponse
+        }
+        // Create a new reader for the response body
+        resp.Body = io.NopCloser(bytes.NewBuffer(body))
+        
+        harResponse.Content = Content{
+            Size:     len(body),
+            Text:     string(body),
+            MimeType: resp.Header.Get("Content-Type"),
+        }
+    }
+
+    return &harResponse
+}
+
+func parseContent(resp *http.Response, harContent *Content) {
+	defer func() {
+		if e := recover(); e != nil {
+			log.Printf("Error parsing response to %v: %v\n", resp.Request.URL, e)
+		}
+	}()
+
+	contentType := resp.Header["Content-Type"]
+	if contentType == nil {
+		panic("Missing content type in response")
+	}
+	harContent.MimeType = contentType[0]
+	if resp.ContentLength == 0 {
+		log.Println("Empty content")
+		return
+	}
+
+	body, _ := ioutil.ReadAll(resp.Body)
+	harContent.Text = string(body)
+	harContent.Size = len(body)
+	return
+}
```

Comment:
This function is not used, should we remove it?

Reply:
I thought it should be kept in for future development, but sure I'll remove it

Reply:
Its functionality is already implemented inside another function in the same file, it can be safely removed, probably it's legacy code